routemaster/models/event.rb

```diff
@@ -12,24 +13,26 @@ class Event < Message
 
       VALID_TYPES = %w(create update delete noop)
 
-      attr_reader :topic, :type, :url
+      attr_reader :topic, :type, :url, :data
 
       def initialize(**options)
         super
-        _assert VALID_TYPES.include?(options[:type]), 'bad event type'
+        @type      = options.fetch(:type, nil)
+        @url       = CallbackURL.new options.fetch(:url, nil)
```

Comment:
I wonder whether callback URL should be allowed to be nil even when there is a payload? Even things that have payloads should be resourceful, and so there should still be a location for it, even if the location is only for the latest value. I see 'data' as being a convenience for fast-changing resources rather than a means to make things non-RESTful. In any case, it should be mandatory when there is no data otherwise the event is useless.